sources/packages/backend/apps/api/src/route-controllers/overaward/overaward.controller.service.ts

```diff
@@ -0,0 +1,73 @@
+import { Injectable } from "@nestjs/common";
+import { DisbursementOverawardService } from "@sims/services";
+import { DisbursementOveraward } from "@sims/sims-db";
+import { getUserFullName } from "../../utilities";
+import {
+  OverawardAPIOutDTO,
+  OverawardBalanceAPIOutDTO,
+} from "./models/overaward.dto";
+
+/**
+ * Overaward controller service.
+ */
+@Injectable()
+export class OverawardControllerService {
+  constructor(
+    private readonly disbursementOverawardService: DisbursementOverawardService,
+  ) {}
+
+  /**
+   * Get the overaward balance of a student.
+   * @param studentId student.
+   * @returns overaward balance for student.
+   */
+  async getOverawardBalance(
+    studentId?: number,
+  ): Promise<OverawardBalanceAPIOutDTO> {
+    const overawardBalance =
+      await this.disbursementOverawardService.getOverawardBalance([studentId]);
+
+    return { overawardBalanceValues: overawardBalance[studentId] };
+  }
+
+  /**
+   * Get all overawards which belong to a student.
+   * @param studentId student.
+   * @param includeAddedBy include added by.
+   * @returns overaward details of a student.
+   */
+  async getOverawardsByStudent(
+    studentId?: number,
+    includeAddedBy = false,
```

Comment:
Maybe we can move this to an option parameter and rename it to `audit`.

Reply:
if we have it as optional parameter, we may need to pass in both the methods, rather defaulting it to false controls the default behaviour and also give us an option to add when its needed. So kept it that way.

Reply:
@guru-aot, please see the comment added by @ann-aot.
Even following that the method can still assign a default value if the `options?.something` is not provided.
We have been doing it many times in the past PRs.
It can be considered a suggestion, but I believe that it is more aligned with what the Team has been doing lately.

Reply:
I understand the use of using options, if the parameter passed is single, and no use of any expansion in the near future. If we are adding options for consistency, I can go ahead and add them, but its again an additional minor overhead when we are translating the structure rather passing the parameter for a single one.